File: 10.Python/Armstrong.py
import math
def main():
    iCount=0
    iRem=0
    iSum=0
    No=0
    iCopy=0

    print("Enter the number:")
    No=int(input())

    iCopy=No
    while iCopy!=0:
        iCount+=1
        iCopy=math.floor(iCopy/10)

    iCopy=No

    while iCopy!=0:
        iRem=iCopy%10
        iSum+=pow(iRem,iCount)
        iCopy=math.floor(iCopy/10)

    if No==iSum:
        print("Armstrong")
    else:
        print("Not Armstrong")

if __name__ =="__main__":
    main()

File: 0.C_CPP_JAVA Programs/JAVA/Armstrong.java
import java.lang.*;
import java.util.Scanner;
public class Armstrong
{
    public static void main(String[] args)
    {
        int iCount=0,iCopy=0,iRem=0,iSum=0;
        Scanner reader=new Scanner(System.in);
        System.out.println("Enter the number");
        int iNo=reader.nextInt();

        iCopy=iNo;

        while(iCopy!=0)
        {
            iCount++;
            iCopy=iCopy/10;
        }
        iCopy=iNo;

        while(iCopy!=0)
        {
            iRem=iCopy%10;
            iSum+=Math.pow(iRem,iCount);
            iCopy=iCopy/10;
        }
        if(iSum==iNo)
        {
            System.out.println("Armstong");
        }
        else
        {
            System.out.println("Non Armstrong");
        }
    }
}